app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,24 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    if @post.save
+      redirect_to "/users/#{current_user.id}/posts"
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] I am unable to create a post. There are validation errors related comments and likes counters. Kindly fix that.

  #### SUGGESTION:
  Initialize your `@post` variable to include the counters:
  ```
  @post = Post.new(post_params.merge!(likes_counter: 0, comments_counter: 0))
  ```
  ![image](https://user-images.githubusercontent.com/81468675/225291034-ba0e7014-eb9b-4f08-8854-7c1547461784.png)